### ISSUE TITLE
Fails to build from source on 32-bit platforms

### ISSUE THREAD

**Issue by ptrcnull** [juanfont/headscale#704]
**Bug description**

fails to build from source on 32-bit platforms:
```
# github.com/puzpuzpuz/xsync
/home/buildozer/go/pkg/mod/github.com/puzpuzpuz/xsync@v1.2.1/util.go:20:24: 0xff51afd7ed558ccd (untyped int constant 18397679294719823053) overflows uintptr
/home/buildozer/go/pkg/mod/github.com/puzpuzpuz/xsync@v1.2.1/util.go:21:24: 0xc4ceb9fe1a85ec53 (untyped int constant 14181476777654086739) overflows uintptr
```

**To Reproduce**

```
$ GOARCH=386 go build
```

**Proposed solution**
can be fixed by bumping xsync version to 1.4.1, where this was [fixed][1]; it doesn't look like there have been any breaking changes since 1.2.1: https://github.com/puzpuzpuz/xsync/releases

[1]: https://github.com/puzpuzpuz/xsync/releases/tag/v1.4.1

**Comment by juanfont** [comment]
This has been fixed in HEAD in #722.

**Comment by ptrcnull** [comment]
thanks!
